src/routing.rs

```diff
@@ -274,12 +276,16 @@ where
                 }
             })
             .map(|href| {
-                event.prevent_default(); // Prevent page refresh
                 // @TODO should be empty href ignored?
-                if !href.is_empty() {
+                if href.is_empty() {
+                    event.prevent_default(); // Prevent page refresh
+                } else {
                     // Route internally based on href's value
                     let url = push_route(Url::from(href));
```

Comment:
Why is `push_route` outside of the condition below?
(Why not something like this:
```rust
let url = Url::from(href);
if let Some(redirect_msg) = routes(url) {
    push_route(url);
    event.prevent_default(); // Prevent page refresh
    update(redirect_msg);
}
```
?)
I don't know History API so well, I'm just curious.

Reply:
Hmmm, wasn't really thinking about it when I added the change., I was just trying to keep behavior as consistent as possible when modifying the code.

Reply:
Wait, I totally misremembered my code. I will look into it. I think it was because I wasn't thinking about history. Ooops.

Reply:
You were right about `push_route`! The history buttons finally work again.